src/DimpleB/StringArrayProblem1.java

```diff
@@ -0,0 +1,71 @@
+/* 1)	WAP which fulfill below condition
+a.	WAP to find positive and negative number from the given array  
+b.	check the sum of individual.
+Input array;- {5, 6, -9, 14, -25, 6, 7, -14}
+Output :- expected output of
+a.	positive number = {5, 6, 14, 6, 7}
+b.	negative number = {-9, -25, -14}
+c.	positive number sum :- 38
+d.	negative number sum :- 48
+*/
+
+import java.util.Scanner;
+
+public class StringArrayProblem1 {
+	static void printsum(int input[])
+	{
+		int i,sumpos=0,sumneg=0,negcnt=0,poscnt=0;
+		int posarr[]=new int[input.length];
+		int negarr[]=new int[input.length];
+		for(i=0;i<input.length;i++)
+		{
+			if(input[i]<0)
+			{
+				sumneg=sumneg+input[i];
+				negarr[i]=input[i];
+				poscnt++;
+				
+			}
+			else if(input[i]>0)
+			{
+				sumpos=sumpos+input[i];
+				posarr[i]=input[i];
+				negcnt++;
+			}
+		}
+		System.out.println("Sum of positive elements:"+sumpos);
+		System.out.println("Sum of negative elements:"+sumneg);
+		System.out.println("Positive count:"+poscnt);
+		System.out.println("Negative count:"+negcnt);
+		System.out.println("Array containing positive elements: ");
+		for(i=0;i<posarr.length;i++)
+		{
+			if(posarr[i]!=0)
+				System.out.print(posarr[i]+"   ");
+		}
+		System.out.println();
+		System.out.println("Array containing negative elements: ");
+		for(i=0;i<negarr.length;i++)
+		{
+			if(negarr[i]!=0)
+			System.out.print(negarr[i]+"   ");
+		}
+	}
+	public static void main(String[] args) {
+		// TODO Auto-generated method stub
+		int i;
```

Comment:
Here you can write separate. Ethos which gives you count of all positive n negative number... On that basis you initialized array Length....
This suggestion to improve your code quality..